src/Symfony/Component/Cache/Traits/RedisTrait.php

```diff
@@ -182,10 +182,25 @@ public static function createConnection($dsn, array $options = array())
 
             $redis = $params['lazy'] ? new RedisClusterProxy($initializer) : $initializer();
         } elseif (is_a($class, \Predis\Client::class, true)) {
-            $params['scheme'] = $scheme;
-            $params['database'] = $params['dbindex'] ?: null;
-            $params['password'] = $auth;
-            $redis = new $class((new Factory())->create($params));
+            if ('server' === $params['cluster']) {
+                $host = $params['host'];
+                if (isset($params['port'])) {
+                    $host .= ':'.$params['port'];
+                }
+
+                $host = array_map(function (string $host) use ($scheme): string {
+                    return $scheme.'://'.$host;
+                }, explode(',', $host));
+
+                // Predis cluster only supports an array of hosts as first argument, otherwise
+                // options array is ignored.
+                $redis = new $class($host, array('cluster' => 'redis'));
```

Comment:
I believe `cluster=redis` should be based on the `cluster` dsn parameter. If the dsn parameter is `server`, then here it should be redis`, if the dsn parameter is `client`, then here it should be `predis`.

Reply:
Like I said in a previous comment, the client part has been not implemented as could involve some other options to be handled, so I think it should be done and discussed in another PR